components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/impl/ActionExecutorServiceImpl.java

```diff
@@ -472,11 +529,14 @@ private String serializeFailureResponse(ActionInvocationFailureResponse response
     }
 
     private List<PerformableOperation> validatePerformableOperations(
-            ActionExecutionRequest request, ActionInvocationSuccessResponse response, Action action) {
+            ActionExecutionRequest request, List<PerformableOperation> operations, Action action) {
 
         List<AllowedOperation> allowedOperations = request.getAllowedOperations();
 
-        List<PerformableOperation> allowedPerformableOperations = response.getOperations().stream()
+        if (operations == null) {
+            return new ArrayList<>();
+        }
```

Comment:
why create a new list. This method just accepts a list so can we rather hold to the contract that it expects a nullable list ? Or can this just return

Reply:
For custom authentication, this will be null. Will remove this part for now, and check for better solution with the effort of adding data attribute for the ActionInvocationSuccessResponse.